Game/AI/Decks/YubelExecutor.cs

```diff
@@ -0,0 +1,2315 @@
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+        public int CompareUsableAttack(ClientCard cardA, ClientCard cardB)
+        {
+            if (cardA == null && cardB == null)
+                return 0;
+            if (cardA == null)
+                return -1;
+            if (cardB == null)
+                return 1;
+            int powerA = (cardA.IsDefense() && summonThisTurn.Contains(cardA)) ? 0 : cardA.Attack;
+            int powerB = (cardB.IsDefense() && summonThisTurn.Contains(cardB)) ? 0 : cardB.Attack;
+            if (powerA < powerB)
+                return -1;
+            if (powerA == powerB)
+                return CardContainer.CompareCardLevel(cardA, cardB);
+            return 1;
+        }
+        private bool ActRageQuickLink()
+        {
+            if (Card.Location != CardLocation.MonsterZone) return false;
+            // Quick Link เฉพาะเทิร์นศัตรูช่วง Main เพื่อสร้าง 1 interrupt
+            if (Duel.Player != 1) return false;
+            if (Duel.Phase < DuelPhase.Main1 || Duel.Phase > DuelPhase.Main2) return false;
+            if (!HasInExtra(CardId.SP_LITTLE_KNIGHT)) return false;
+
+            var target = GetBestEnemyMonster(onlyFaceup: true, canBeTarget: true);
+            if (target == null) return false;
+
+            // ส่วนใหญ่เอนจินจะถามเลือกการ์ดฝั่งตรงข้ามที่จะใช้เป็นวัสดุ/หรือไล่
+            AI.SelectCard(target);
+            return DontSelfNG();
+        }
+        private bool SSPhantom()
+        {
+            ClientCard mat2;
+            var gyList = Bot.Graveyard.ToList();
+            // ต้องมี Terror อยู่ในมือหรือสุสานก่อน
+            if (Bot.HasInMonstersZone(CardId.YUBEL_TERROR_INCARNATE))
+            {
+                if (Bot.HasInMonstersZone(CardId.YUBEL_TERROR_INCARNATE))
+                    AI.SelectCard(CardId.YUBEL_TERROR_INCARNATE);
+                else
+                    return false;
+            }
+            else if (Bot.HasInHandOrInGraveyard(CardId.YUBEL_TERROR_INCARNATE))
+            {
+                // วัตถุดิบใบที่ 2: หาจากสุสานเท่านั้น (ยกเว้น Terror)
+                gyList = Bot.Graveyard
+                .GetMatchingCards(c => c != null && c.IsMonster() && c.Id != CardId.YUBEL_TERROR_INCARNATE)
+                .ToList();
+                // วัตถุดิบใบที่ 1: Terror (ให้เลือกจากสุสานก่อน ถ้าไม่มีค่อยเลือกจากมือ)
+                if (Bot.HasInGraveyard(CardId.YUBEL_TERROR_INCARNATE))
+                {
+                    AI.SelectCard(CardId.YUBEL_TERROR_INCARNATE);
+                }
+                else if (Bot.HasInHand(CardId.YUBEL_TERROR_INCARNATE))
+                    AI.SelectCard(CardId.YUBEL_TERROR_INCARNATE);
+                else
+                    return false;
+
+            }
+            else if (Bot.HasInMonstersZone(CardId.YUBEL))
+            {
+                if (Bot.HasInMonstersZone(CardId.YUBEL))
```

Comment:
Same as above. And those checks won't cover the all circumstances, so the select may be inaccurate.